beamer/contracts.py

```diff
@@ -94,10 +94,19 @@ def load_deployment_info(artifacts_dir: Path, abi_dir: Path) -> DeploymentInfo:
     return deployment_info
 
 
-def contracts_for_web3(web3: Web3, artifacts_dir: Path, abi_dir: Path) -> dict[str, Contract]:
-    deployment_info = load_deployment_info(artifacts_dir, abi_dir)
-    chain_id = ChainId(web3.eth.chain_id)
-    return make_contracts(web3, deployment_info[chain_id])
+def contracts_for_web3(w3: Web3, artifacts_dir: Path, abi_dir: Path) -> dict[str, Contract]:
```

Comment:
Well this is not like super accurate because w3 can be basically l1 and then you need to traverse and get l1messengers as well to be more accurate

Reply:
I'd like to chime in here. looking at `contracts_for_web3` this function is only used in `call_contracts`,`e2e-test-fill` and `e2e-test-verify`.

Speaking for the e2e tests at least, we only need RequestManager FillManager and MintableToken. 
How far would it be away to also use the new API in the tests and get rid of contracts_for_web3 completely? It doesn't seem to have too much use in our code base.

Reply:
Looked at call_contracts, here we are also only ever using the above mentioned contracts

Reply:
> I'd like to chime in here. looking at `contracts_for_web3` this function is only used in `call_contracts`,`e2e-test-fill` and `e2e-test-verify`.
> 
> Speaking for the e2e tests at least, we only need RequestManager FillManager and MintableToken. How far would it be away to also use the new API in the tests and get rid of contracts_for_web3 completely? It doesn't seem to have too much use in our code base.

This is precisely what I was debating myself before finishing the branch. `contracts_for_web3` is primarily used as a convenience function for scripts like the ones you named. In this light, it can come quite handy if you don't want to deal with loading chain-specific artifacts on your own. Also, given that it constructs an empty `ABIManager` every time it is called, it is clearly not ideal in case of multiple calls; I've even wanted to add a comment regarding that, along the lines of `to be used only in scripts`.

On the other hand, I think @bilbeyt has a valid point here and I would like to find a better solution.
Also, `beamer.artifacts.load_all` is not ideally named and I would like to improve on that.
Perhaps with an additional API to help out "the script use cases" and better naming we can get rid of this function entirely.

Reply:
Something like this, perhaps?
```diff
diff --git a/scripts/e2e-test-fill.py b/scripts/e2e-test-fill.py
index b0ceede8..394e9278 100644
--- a/scripts/e2e-test-fill.py
+++ b/scripts/e2e-test-fill.py
@@ -20,15 +20,17 @@ def main() -> None:
     deployer = account_from_keyfile(keystore_file, password)
     web3 = make_web3(l2_rpc, deployer)
 
-    l2_contracts = contracts_for_web3(web3, artifacts_dir, abi_dir)
+    abi_manager = ABIManager(abi_dir)
+    deployment = beamer.artifacts.load(web3.eth.chain_id)
+
     overriden_chain_id = os.getenv("SOURCE_CHAIN_ID")
     if overriden_chain_id is None:
         source_chain_id = ChainId(web3.eth.chain_id)
     else:
         source_chain_id = ChainId(int(overriden_chain_id))
 
-    fill_manager = l2_contracts["FillManager"]
-    token = l2_contracts["MintableToken"]
+    fill_manager = obtain_contract(web3, abi_manager, deployment, "FillManager")
+    token = obtain_contract(web3, abi_manager, deployment, "MintableToken")
 
     nonce = random.randint(1, sys.maxsize)
     request_amount = 123
```

Reply:
I have pushed a changes like above and removed both `beamer.artifacts.load_all` and `beamer.contracts.contracts_for_web3`, I think now it makes much more sense. Please take a look again.

Reply:
looks much better 👍